app/controllers/songs_controller.rb

```diff
@@ -1,10 +1,25 @@
 class SongsController < ApplicationController
   def index
+    @songs = Song.all
   end
 
   def new
   end
 
   def edit
   end
+  
+  def search
+    @results = true
+    if params[:search].nil?
+      @songs = Song.all
+    else
+      @songs = Song.search params[:search]
+      if @songs.nil? or @songs.none?
```

Comment:
Don't use `or`, use `||`. http://devblog.avdi.org/2010/08/02/using-and-and-or-in-ruby/